src/functions.c

```diff
@@ -372,6 +375,154 @@ void fcallroCommand(client *c) {
     fcallCommandGeneric(c, 1);
 }
 
+/*
+ * FUNCTION DUMP
+ *
+ * Returns a binary blob representing all the functions.
+ * Can be loaded using FUNCTION RESTORE
+ */
+void functionDumpCommand(client *c) {
+    unsigned char buf[2];
+    uint64_t crc;
+    rio payload;
+    rioInitWithBuffer(&payload, sdsempty());
+
+    functionsSaveRio(&payload);
+
+    /* RDB version */
+    buf[0] = RDB_VERSION & 0xff;
+    buf[1] = (RDB_VERSION >> 8) & 0xff;
+    payload.io.buffer.ptr = sdscatlen(payload.io.buffer.ptr, buf, 2);
+
+    /* CRC64 */
+    crc = crc64(0, (unsigned char*) payload.io.buffer.ptr,
+                sdslen(payload.io.buffer.ptr));
+    memrev64ifbe(&crc);
+    payload.io.buffer.ptr = sdscatlen(payload.io.buffer.ptr, &crc, 8);
+
+    addReplyBulkSds(c, payload.io.buffer.ptr);
+}
+
+/*
+ * FUNCTION RESTORE <blob> [FLUSH|APPEND|REPLACE]
+ *
+ * Restore the functions represented by the give blob.
+ * Restore policy to can be given to control how to handle existing functions:
+ * * FLUSH: delete all existing functions.
+ * * APPEND: appends the restored functions to the existing functions. On collision, abort.
+ * * REPLACE: appends the restored functions to the existing functions.
+ *   On collision, replace the new function with the old function.
+ */
+void functionRestoreCommand(client *c) {
+#define RESTORE_POLICY_FLUSH 1
+#define RESTORE_POLICY_APPEND 2
+#define RESTORE_POLICY_REPLACE 3
+    if (c->argc > 4) {
+        addReplySubcommandSyntaxError(c);
+        return;
+    }
+
+    int restore_replicy = RESTORE_POLICY_FLUSH; /* default policy: FLUSH */
```

Comment:
Are you sure FLUSH should be the default? 
Maybe MERGE is the safest. 
That's the only one that doesn't delete old data (it errors instead).. 

Either way, let's document that in the top comment and maybe HELP message

Reply:
Changed default to append.